apps/zk-explorer/containers/RelatedProjectsContainer/client/RelatedProjectsCarousel.tsx

```diff
@@ -0,0 +1,66 @@
+'use client';
+
+import { type FC, ComponentProps, useMemo } from 'react';
+import Link from 'next/link';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import { Pagination } from 'swiper';
+import 'swiper/css';
+import 'swiper/css/pagination';
+
+import { ProjectCard } from '../../../components/ProjectCard/ProjectCard';
+import type { ProjectItem } from '../../../components/ProjectCard/types';
+
+const RelatedProjectsCarousel: FC<{ projects: ProjectItem[] }> = ({
+  projects,
+}) => {
+  const sharedSwiperProps = useMemo(
```

Comment:
We don't really need `useMemo` here since the value is static and calculation of this value is not dependant on anything. `sharedSwiperProps` can also be defined outside the component without the use of `useMemo` like this:

```
 const sharedSwiperProps: ComponentProps<typeof Swiper> = {
    spaceBetween: 20,
    pagination: {
      clickable: true,
    },
    navigation: true,
    modules: [Pagination],
  };
```